gpt4all-backend/llamamodel.cpp

```diff
@@ -136,13 +138,20 @@ static gguf_context *load_gguf(const char *fname) {
 }
 
 static int32_t get_arch_key_u32(std::string const &modelPath, std::string const &archKey) {
+    int32_t value = -1;
+    std::string arch;
+
     auto * ctx = load_gguf(modelPath.c_str());
     if (!ctx)
-        return -1;
-    std::string arch = get_arch_name(ctx);
+        goto cleanup;
 
-    int32_t value = -1;
-    if (ctx) {
+    try {
```

Comment:
In general, I prefer for one line conditionals not to have braces unless it aids with code readability. In general, try and err on the side of not including the braces if they don't help and refrain from adding braces if they don't help.